utils/neuralmagic/sparse_train_manager.py

```diff
@@ -74,6 +74,39 @@ def __init__(
                 self.model, last_epoch if last_epoch >= 0 else float("inf")
             )
 
+    def check_for_invalid_state(self):
+        """
+        Checks that the training sparsification recipe (or lack of) is a valid recipe
+        for the loaded model. This primarily applies when the loaded model is already
+        sparsified.
+        """
+
+        # Checking valid state for pruned models
+        if self.checkpoint_manager and self.checkpoint_manager.pruning_modifiers:
+            if not self.train_manager:
+                self.log_console(
+                    "Model in danger of de-sparsification. Pruned model was loaded, "
```

Comment:
leaning towards taking out `Model in danger of de-sparsification.`. LGTM otherwise